### ISSUE TITLE
[BUG] CUDF crashes trying to  read a valid ORC file with a single null entry in it

### ISSUE THREAD

**Issue by revans2** [rapidsai/cudf#12155]
**Describe the bug**
The file that causes this error is attached below

**Steps/Code to reproduce bug**
Try to read this particular file using CUDF.  It will throw an exception like

```
cuDF failure at: .../cpp/src/io/orc/reader_impl.cu:1081: Expected streams data within stripe
```

I am not an expert in ORC, but I tried to debug this a little bit at the point in which the error is being thrown we are trying to read two STRING columns (I assume the key and value for the map).

**Expected behavior**
I should be able to read the file and get the null out, like I can in Spark.

```
scala> spark.read.orc("503819761.orc").show()
+---------+
|query_map|
+---------+
|     null|
+---------+

```
or pandas

```
>>> pd.read_orc("503819761.orc")
  query_map
0      None
```

**Comment by revans2** [comment]
[error.zip](https://github.com/rapidsai/cudf/files/10016724/error.zip) has the offending ORC file in it.  Jut FYI I also found another error when trying to write a separate repro case that I will file a separate issue for. It might be related.

**Comment by revans2** [comment]
[combined_error.zip](https://github.com/rapidsai/cudf/files/10023410/combined_error.zip) Adding a second file that is like the first file, but also has a few more rows. This is closer to what our customer is seeing in production.

**Comment by vuule** [comment]
Closed via #12160
